### PR TITLE
Add upgrade support of labels/taints for Docker

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/1626

*Description of changes:*
Add support for upgrading labels and taints for the Docker provider which was missing before.

*Testing (if applicable):*
Ran e2e tests while running tilt to update the controller

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/vsphere/vsphere.go [hunk 1]
```diff
@@ -884,10 +884,6 @@ func (p *vsphereProvider) generateCAPISpecForUpgrade(ctx context.Context, bootst
 			return nil, nil, err
 		}
 		needsNewWorkloadTemplate, err := p.needsNewMachineTemplate(currentSpec, newClusterSpec, workerNodeGroupConfiguration, vdc, previousWorkerNodeGroupConfigs, oldWorkerNodeVmc, newWorkerNodeVmc)
-		if err != nil {
```
Comment (danbudris) [inline pkg/providers/vsphere/vsphere.go:887]: how did this get here ha

**Review by danbudris — APPROVED** [review]
/lgtm

File: pkg/providers/vsphere/vsphere.go (hunk 1 above)
Comment (vivek-koppuru) [reply pkg/providers/vsphere/vsphere.go:887]: No idea haha

**Pull request merged**